File: Haskell/Services/CriarLocalService.hs
module Services.CriarLocalService (
    criarLocalService
) where

import Local.Local (criarLocal)
import Local.Agenda (geraAgenda)

criarLocalService :: String -> [String] -> Int -> IO ()
criarLocalService nome recursos capacidade = do
    criarLocal nome recursos capacidade
    geraAgenda nome

File: Prolog/Services/Local/CriarLocalService.pl
:- module(CriarLocalService, [criarLocal/2]).
:- use_module('../../Models/Local/ModelLocal.pl').
:- use_module('../../Repository/localRepository.pl').

criarLocal(Nome, Materiais) :-
    createLocal(Nome, Materiais, Local),
    saveLocal(Local).